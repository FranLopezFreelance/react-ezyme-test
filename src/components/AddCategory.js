import React, { useState } from 'react';
import PropTypes from 'prop-types';

const AddCategory = ({setCategory}) => {
  
  const [inputValue, setInputValue] = useState('');

  const handleInputChange = (e) => {
    setInputValue(e.target.value)
  }

  const handleAddCategory = () => {
    if(inputValue.trim().length > 1){
      setCategory(cats => [inputValue, ...cats ]);
      setInputValue('');
    }
  }

  const handleSubmit = (e) => {
    e.preventDefault();
    if(inputValue.trim().length > 1){
      setCategory(cats => [inputValue, ...cats ]);
      setInputValue('');
    }
  }

  return (
    <form onSubmit={handleSubmit}>
      <input 
        placeholder="Buscar... "
        type="text" 
        className="category-input" 
        value={inputValue} 
        onChange={handleInputChange} />
      <button 
        className="add-category"
        onClick={handleAddCategory}>Buscar</button>
    </form>
  )
}

AddCategory.propTypes = {
  setCategory: PropTypes.func.isRequired
}

export default AddCategory
