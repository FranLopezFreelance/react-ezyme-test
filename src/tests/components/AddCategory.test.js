import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';

import  AddCategory from '../../components/AddCategory';

describe('Pruebas en <AddCategory />', () => {

  const setCategory = jest.fn();
  let wrapper;
  beforeEach(() => {
    jest.clearAllMocks();
    wrapper = shallow(<AddCategory setCategory={setCategory} />);
  });

  test('debe mostrar el componente correctamente', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('Debe de cambiar el input', () => {
    const input = wrapper.find('input');
    const value = 'Hola mundo';
    input.simulate('change', {target:{ value }});
  });

  test('No debe postear la informaciÃ³n onSubmit', () => {
    wrapper.find('form').simulate('submit', { preventDefault(){} });  
    expect(setCategory).not.toHaveBeenCalled();
  });

  test('Debe llamar el setCategory y limpiar caja de texto', () => {
    const input = wrapper.find('input');
    const value = 'Hola mundo';
    input.simulate('change', {target:{ value }});
    wrapper.find('form').simulate('submit', { preventDefault(){} });  
    /* expect(setCategory).toHaveBeenCalled(); */
    expect(setCategory).toHaveBeenCalledTimes(1);
    expect(setCategory).toHaveBeenCalledWith( expect.any(Function) );
    expect(input.prop('value')).toBe('');
  });
});