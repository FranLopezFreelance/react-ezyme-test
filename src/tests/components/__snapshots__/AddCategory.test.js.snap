// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <AddCategory /> debe mostrar el componente correctamente 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    className="category-input"
    onChange={[Function]}
    placeholder="Buscar... "
    type="text"
    value=""
  />
  <button
    className="add-category"
    onClick={[Function]}
  >
    Buscar
  </button>
</form>
`;
